### DISCUSSION TITLE
Send Response and then "Work"

### DISCUSSION THREAD

**Discussion by cldparisi** [nextgenhealthcare/connect#5525] [General]
Hi,
I have these needs:

- create a HTTP Listener
- client call HTTP Listener
- HTTP listener has to do a long job so it immediately sends a rresponse to client and only then it works

I thought it was enough to set "Wait for previous destination" to false on Response step but it doesn't work.
Response is created first but it is not send.

![image](https://user-images.githubusercontent.com/43777441/201318355-b255b846-d1d4-4b25-8456-59fbe278e11b.png)

XPOST: https://forums.mirthproject.io/forum/mirth-connect/support/182517-send-response-and-then-work

**Comment by jonbartels** [comment]
Build your response in the source transformer and do `responseMap.put('doing_work', res)` then you will be able to set your source connector to "Respond From:" to "doing_work". Also set your Long Operation destination to "Queue always".

The order of operations is:
1. Source connector gets message
2. Source connector runs source transformers
3. Source transformer builds the response
4. Source connector puts the message to the destination queue for your Long Operation destination
5. Source connector responds to the caller with the "doing_work" response
6. Destination connector for Long Operation reads its queue
7. Destination connector for Long Operation does transformations and sends the message
